src/gnssro/gnss_tec_ground_based.py

```diff
@@ -0,0 +1,469 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2019-2024 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+#
+# decode TENET formatted ground-based GNSS Total Electron Content (TEC)
+#
+
+import sys
+import os
+import time
+from datetime import datetime, timezone
+import numpy as np
+import logging
+
+# These modules need the path to lib-python modules
+import pyiodaconv.ioda_conv_engines as iconv
+from pyiodaconv.orddicts import DefaultOrderedDict
+from pyiodaconv.def_jedi_utils import iso8601_string, epoch
+
+os.environ["TZ"] = "UTC"
+
+# these are the unique values in the raw input file
+varDict = {'totalElectronContentVertical': ['totalElectronContentVertical', "integer", 'TECU'],
+           'totalElectronContentSlant': ['totalElectronContentSlant', "integer", "TECU"],
+}
+
+# these are the MetaData common to each input
+locationKeyList = [
+    ('latitude', 'float', 'degrees_north'),
+    ('longitude', 'float', 'degrees_east'),
+    ('pseudoRandomNoiseCode', 'integer', 'GNSS transmitter pseudoRandomNoise PRN code'),
```

Comment:
this is potentially another new one @haydenlj , need to look what we have for GNSS already the PRN codes are for each GNSS constellation

Reply:
@BenjaminRuston looks like we don't have anything for PRN specifically, what we have is satellite identifier https://github.com/JCSDA-internal/ioda/blob/develop/share/ioda/yaml/validation/ObsSpace.yaml#L887
Is this different?

Reply:
the PRN number is kinda like the identifier number or series number from each constellation.... so GPS has 1-42 PRN,,, and so does Galileo and GLONASS.

in the conventions we have the `satelliteTransmitterId` which I think is what I'm looking for:
https://github.com/JCSDA-internal/ioda/blob/24df6cc5938d655aed1274af5605b53795dd55bb/share/ioda/yaml/validation/ObsSpace.yaml#L890

and then this defined GPS, GLONASS, Galileo, ... :
https://github.com/JCSDA-internal/ioda/blob/24df6cc5938d655aed1274af5605b53795dd55bb/share/ioda/yaml/validation/ObsSpace.yaml#L896

does this make sense to you, think this is the correct one to use... we can ask others if it makes sense too

Reply:
Yes that makes sense

Reply:
I just looked at a IODA file and yes this is what we want:

```
range(/MetaData/satelliteTransmitterId): [1, 32]
```